ext/intl/dateformat/dateformat_create.cpp

```diff
@@ -113,7 +113,8 @@ static zend_result datefmt_ctor(INTERNAL_FUNCTION_PARAMETERS, zend_error_handlin
 	locale = Locale::createFromName(locale_str);
 	/* get*Name accessors being set does not preclude being bogus */
 	if (locale.isBogus() || strlen(locale.getISO3Language()) == 0) {
-		goto error;
+		intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR, "datefmt_create: invalid locale", 0);
+		return FAILURE;
```

Comment:
```suggestion
		zend_argument_value_error(1, "\"%s\" is invalid", locale);
		return FAILURE;
```

Apply the OO error style to procedural?